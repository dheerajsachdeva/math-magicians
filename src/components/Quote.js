import { useState, useEffect } from 'react';

const Quote = () => {
  const [data, setData] = useState([]);
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      try {
        const apiKey = 'k8Y//jwNVrEiV+8XyecTdQ==dbnY7z0CBmyoJMiC';
        const options = {
          method: 'GET',
          headers: {
            'Content-Type': 'application/json',
            'X-Api-Key': apiKey,
          },
        };
        const url = 'https://api.api-ninjas.com/v1/quotes?category=happiness';
        const response = await fetch(url, options);
        const text = await response.json();
        setData(text[0]);
      } catch (error) {
        setError(true);
      }
      setLoading(false);
    };
    fetchData();
  }, [setData, setLoading]);

  if (loading) return <div className="loading">Loading...</div>;
  if (error) return <div className="error">Something went wrong!</div>;

  return (
    <>
      <div>{loading}</div>
      <div>{error}</div>
      <div data-testid="quote-test" className="quote">
        Quote :
        {data.quote}
        {' '}
        <br />
        {' '}
        By :
        {' '}
        {data.author}
      </div>
    </>
  );
};

export default Quote;
