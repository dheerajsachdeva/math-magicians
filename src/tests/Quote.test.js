import { render, act } from '@testing-library/react';
import Quote from '../components/Quote';

global.fetch = require('jest-fetch-mock');

beforeEach(() => {
  fetch.resetMocks();
  fetch.mockResponse(JSON.stringify([{ quote: 'Testing Quote', author: 'Kevin and Dheeraj' }]));
});

it('quote render OK', async () => {
  const quote = await act(async () => render(<Quote />));

  expect(quote).toMatchSnapshot();
});
