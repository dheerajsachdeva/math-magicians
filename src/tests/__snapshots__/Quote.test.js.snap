// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div />,
  <div />,
  <div
    className="quote"
  >
    Quote :
     
    <br />
     
    By :
     
  </div>,
]
`;
